crates/polars-core/src/utils/mod.rs

```diff
@@ -740,10 +740,21 @@ where
     let mut iter = dfs.into_iter();
     let additional = iter.size_hint().0;
     let mut acc_df = iter.next().unwrap();
+    let mut acc_height = acc_df.height();
     acc_df.reserve_chunks(additional);
     for df in iter {
+        acc_height += df.height();
         acc_df.vstack_mut(&df)?;
     }
+
+    if acc_height > IdxSize::MAX as usize {
```

Comment:
I don't think we should check this here. This should fail deeper down when constructing the chunkedarray. The `append` operation should do the height check.

Reply:
We still need some checks for the Parquet reader because of the `row_offset`, but I added the checks to `append` and `extend`.

The following now also throws a proper error:

```python
import polars as pl
import numpy as np

t1 = pl.Series(np.zeros(4_000_000_000), dtype=bool)
t2 = pl.Series(np.zeros(1_000_000_000), dtype=bool)

t1.append(t2)
```

Reply:
Yes, that's better. 

Why do we still need it in parquet? As processing datasets that exceed that limit is fine, we only shouldn't create morsels that are larger.

Reply:
The problem is that `previous_row_count` will silently overflow, which might produce wrong results. If you read a file with 5 billion times `False`. It will overflow there instead of actually getting to the accumulate.

Reply:
And what if we use `u64` here, would that work?

Reply:
It is quite a big change and the overflow will still happen somewhere.

Reply:
Alright. Not worth it for now. :+1: